wip-mariocao-adjust-mining-probability.md

```diff
@@ -0,0 +1,87 @@
+
+A thorough investigation concluded that the majority of these blocks (if not all) were produced by non-reputed and non-active nodes (i.e. newcomers to the network). Regardless of the intentionality of this behaviour, these blocks were appearing in a much higher volume than the newcomer mining probability that was designed in the protocol.
+
+The 18th of January under the following network conditions, the newcomer mining probability was:
+
+TRS(Total Reputation Set)  = 2000 nodes
+ARS(Active Reputation Set) = 6000 nodes
+RF(Replication Factor) = 3
+
+```
+P(non-reputed node mines a block) = P(none reputed node mines a block) = P(single reputed does not mine a block)^TRS = (1- P(single reputed mine a block))^TRS = (1 - RF/ARS)^TRS ~= 37%
+```
+
+In the initial mining algorithm design [rand-poe], DoS attacks were carefully taken into account. An upper bound for the probability of an attacker censoring blocks was set to approximately 2%.
+
+However, before the mainnet launch some mining parameters were changed to achieve a more fair block reward distribution. It was assummed that the reputation expiration would be slower compared to the activity period (related to the ARS). This assumption did not hold in mainnet as seen in the former example. This mistake led to a significant increase in the newcomer mining probability.
+
+Summing up, there is a need for adjusting the newcomer mining probability to mitigate the impact of potential DoS attacks that may endanger the availability and the security of the Witnet network.
+
+
+## Proposal / Specification
+
+The current mining algorithm can be summarized as follows:
+
+1. Miners build a candidate block.
+2. Miners compute their eligibility by using the [VRF]-based [RandPoE] algorithm with by using the ARS census as the difficulty parameter (labelled as N).
+3. If eligible (and no better candidate was received), miners will broadcast their candidate blocks.
+
+Upon reception of multiple candidates, the winner will be selected as follows:
+1. Candidate blocks from reputed nodes are prioritized over those from non-reputed nodes.
+2. If there are still multiple candidates, the one with the lowest VRF is chosen.
+
+The main problem with the aforementioned algorithm is that the eligibility is computed using the ARS as census, but the prioritization is only taking into account the reputed nodes. This led to the increase of the newcomer mining probability.
```

Comment:
This rather belongs in the motivation.

The specification should only tell what the behavior needs to be after adoption of the proposal (_"parte expositiva"_ vs. _"parte dispositiva"_).